server/hap2/hatohol/hap2_nagios_livestatus.py

```diff
@@ -0,0 +1,312 @@
+            })
+
+        if len(result):
+            # livestatus return a sorted list.
+            # result[0] is latest statehist.
+            self.__latest_statehist = json.dumps(result[0])
+        self.put_events(events, fetch_id=fetch_id,
+                        last_info_generator=self.return_latest_statehist)
+
+    def return_latest_statehist(self, events):
+        # livestatus plugins does not use events.
+        # But, haplib.put_events last_info_generator have wanted events.
+        return self.__latest_statehist
+
+    def __parse_status_and_severity(self, status):
+        hapi_status = self.STATUS_MAP.get(status)
+        if hapi_status is None:
+            logger.warning("Unknown status: " + str(status))
+            hapi_status = "UNKNOWN"
+
+        hapi_severity = self.SEVERITY_MAP.get(status)
+        if hapi_severity is None:
+            logger.warning("Unknown status: " + str(status))
+            hapi_severity = "UNKNOWN"
+
+        return (hapi_status, hapi_severity)
+
+
+class Hap2NagiosLivestatusPoller(haplib.BasePoller, Common):
+
+    def __init__(self, *args, **kwargs):
+        haplib.BasePoller.__init__(self, *args, **kwargs)
+        Common.__init__(self)
+
+    def poll_setup(self):
+        self.ensure_connection()
+
+    def poll_hosts(self):
+        self.collect_hosts_and_put()
+
+    def poll_host_groups(self):
+        self.collect_host_groups_and_put()
+
+    def poll_host_group_membership(self):
+        self.collect_host_group_membership_and_put()
+
+    def poll_triggers(self):
+        self.collect_triggers_and_put()
+
+    def poll_events(self):
+        self.collect_events_and_put()
+
+    def on_aborted_poll(self):
+        self.reset()
+
+
+class Hap2NagiosLivestatusMain(haplib.BaseMainPlugin, Common):
+    def __init__(self, *args, **kwargs):
+        haplib.BaseMainPlugin.__init__(self)
+        Common.__init__(self)
+
+    def hap_fetch_triggers(self, params, request_id):
+        self.ensure_connection()
+        # TODO: return FAILURE when connection fails
+        self.get_sender().response("SUCCESS", request_id)
+        fetch_id = params["fetchId"]
+        host_ids = params.get("hostIds")
+        self.collect_triggers_and_put(fetch_id, host_ids)
+
+    def hap_fetch_events(self, params, request_id):
+        self.ensure_connection()
+        # TODO: return FAILURE when connection fails
+        self.get_sender().response("SUCCESS", request_id)
+        self.collect_events_and_put(params["fetchId"], params["lastInfo"],
+                                    params["count"], params["direction"])
+
+
+class Hap2NagiosLivestatus(standardhap.StandardHap):
+    def create_main_plugin(self, *args, **kwargs):
+        return Hap2NagiosLivestatusMain(*args, **kwargs)
+
+    def create_poller(self, *args, **kwargs):
+        return Hap2NagiosLivestatusPoller(self, *args, **kwargs)
+
+
+if __name__ == '__main__':
+    hap = Hap2NagiosLivestatus()
+    hap()
```

Comment:
`hap` has a possibility to be collided for global defined `hap`. see: https://github.com/project-hatohol/hatohol/issues/1877